deps/v8/src/sampler.cc

```diff
@@ -342,6 +342,14 @@ void SignalHandler::HandleProfilerSignal(int signal, siginfo_t* info,
     // We require a fully initialized and entered isolate.
     return;
   }
+
+  // ISSUE-14576: To avoid deadlock, return if there is a thread lock
+  if (Isolate::GetProcessWideMutexPointer()->TryLock()) {
```

Comment:
There are different places where the lock is acquired by the same thread, I've seen different stacks with gdb.

Not sure how/if SIGPROF is blocked, I think not, because it is fired once every millisecond (every 10ms on Android).